test/src/test/java/hudson/search/SearchTest.java

```diff
@@ -225,6 +259,95 @@ else if(displayName.equals(name)) {
         assertTrue(foundDispayName);
     }
 
+    @Issue("JENKINS-24433")
+    @Test
+    public void testProjectNameBehindAFolderDisplayName() throws Exception {
+        final String projectName1 = "job-1";
+        final String displayName1 = "job-1";
+
+        final String projectName2 = "job-2";
+        final String displayName2 = "job-2";
+
+        FreeStyleProject project1 = j.createFreeStyleProject(projectName1);
+        project1.setDisplayName(displayName1);
```

Comment:
What is the point of setting the `displayName` to be the same as the `name`? Do you not mean to test a project with a _different_ display name?